grpc/middleware.go

```diff
@@ -36,3 +37,22 @@ func EncodeContextWithUTMData(parentCtx context.Context) context.Context {
 	}
 	return metadata.AppendToOutgoingContext(parentCtx, utmMetadataKey, w.String())
 }
+
+// AddExternalDependencyMetadataToContext adds external dependencies to the context if they are present in the metadata.
+func AddExternalDependencyMetadataToContext(ctx context.Context, md metadata.MD) context.Context {
```

Comment:
I am a bit confused, why is the function added? Who and where is this supposed to be used?

Reply:
Not confused anymore.